src/main/java/redis/clients/jedis/util/MultiNodePipelineThreadPool.java

```diff
@@ -12,19 +12,68 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import redis.clients.jedis.Protocol;
 
 /**
  * This class is used to build a thread pool for Jedis.
  */
-public class JedisThreadPoolBuilder {
-    private static final Logger log = LoggerFactory.getLogger(JedisThreadPoolBuilder.class);
+public class MultiNodePipelineThreadPool {
```

Comment:
If we name it MultiNodePipelineThreadPool, it is ok at present, but when we need a XXXThreadPool next time, we have to copy the code of MultiNodePipelineThreadPool, which will make a lot of code redundant, such as PoolBuilder.

Reply:
I'm thinking to take the risk.

If such situations comes we can have JedisThreadPool with a name to thread pool map. I'm trying to avoid more complex implementation right now.

Reply:
Can't agree with your point of view. This does not introduce complexity, just add two tool classes `JedisThreadFactoryBuilder` and `JedisThreadPoolBuilder` (putting it in the `util` directory also ok), and then customize the thread pool where needed.